src/main/scala/edu/ie3/util/osm/model/RelationEntitiesSupport.scala

```diff
@@ -0,0 +1,130 @@
+              Map.empty[Long, Node],
+              Map.empty[Long, Way],
+              Map.empty[Long, Relation]
+            ) { case ((nodes, ways, relations), relationMember) =>
+              relationMember.relationType match {
+                case RelationMemberType.Node =>
+                  _nodeEntity(relationMember) match {
+                    case Some(node) =>
+                      (nodes + (node.id -> node), ways, relations)
+                    case None =>
+                      (nodes, ways, relations)
+                  }
+                case RelationMemberType.Way =>
+                  _wayEntity(relationMember) match {
+                    case Some(way) =>
+                      (nodes, ways + (way.id -> way), relations)
+                    case None =>
+                      (nodes, ways, relations)
+                  }
+
+                case RelationMemberType.Relation =>
+                  _relationEntities(relationMember)
+                    .zip(_getRelation(relationMember.id)) match {
+                    case Some((entities, relation)) =>
+                      (
+                        nodes ++ entities.nodes,
+                        ways ++ entities.ways,
+                        relations ++ entities.relations + (relation.id -> relation)
+                      )
+                    case None =>
+                      (nodes, ways, relations)
+                  }
+                case RelationMemberType.Unrecognized =>
+                  logger.warn(
+                    s"RelationMember '$relationMember' has type 'Unrecognized'."
+                  )
+                  (nodes, ways, relations)
+              }
+            }
+            RelationEntities(relationId, nodes, ways, relations)
+          })
+          .map(relationEntities => {
+            _relationEntityCache.putIfAbsent(relationId, relationEntities)
+            relationEntities
+          })
+    }
+  }
+}
+
+object RelationEntitiesSupport {
+
+  /** Class holding all instances of [[Node]] s, [[Way]] s, and [[Relation]] s
+    * inside a specific [[Relation]]. This class is intended to be recursive and
+    * comprehensive. That means, if it holds [[Relation]] instances, all
+    * [[Node]] and [[Way]] instances of all downstream relations are also
+    * included comparable to a deep instance.
+    *
+    * If specific entity instances are not part of the container, they are
+    * (obviously) not included!
+    *
+    * @param relationId
+    *   the identifier of the [[Relation]] this class holds all entities for
+    * @param nodes
+    *   nodes inside the specific relation
+    * @param ways
+    *   ways inside the specific relation
+    * @param relations
+    *   relations inside the specific relation
+    */
+  final case class RelationEntities(
+      relationId: Long,
+      nodes: Map[Long, Node],
+      ways: Map[Long, Way],
+      relations: Map[Long, Relation]
+  )
```

Comment:
Thanks for this point. I also had this solution in mind but thought this would somehow inconsistent bc we have the entity instances on the one hand but only id's for relation on the other hand. Dunno if this might cause some confusion. You might consider handing in an issue for this if you think we should keep track of this. Otherwise we could use this approach if we **really need* kinda extended entity model.